src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```
@@ -0,0 +1,263 @@
+     *
+     * @return true if reads are available, otherwise false
+     */
+    public final boolean hasReads() {
+        return readsSource != null;
+    }
+
+    /**
+     * Are sources of intervals available?
+     *
+     * @return true if intervals are available, otherwise false
+     */
+    public final boolean hasIntervals() {
+        return intervals != null;
+    }
+
+    /**
+     * Window function that controls how much reference context to return for each read when
+     * using the reference source returned by {@link #getReference}. Tools should override
+     * as appropriate. The default function is the identity function (ie., return exactly
+     * the reference bases that span each read).
+     *
+     * @return reference window function used to initialize the reference source
+     */
+    public SerializableFunction<GATKRead, SimpleInterval> getReferenceWindowFunction() {
+        return RefWindowFunctions.IDENTITY_FUNCTION;
+    }
+
+    /**
+     * Returns the "best available" sequence dictionary. This will be the reference sequence dictionary if
+     * there is a reference, otherwise it will be the sequence dictionary constructed from the reads if
+     * there are reads, otherwise it will be null.
+     *
+     * TODO: check interval file(s) as well for a sequence dictionary
+     *
+     * @return best available sequence dictionary given our inputs
+     */
+    public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+        return hasReference() ? referenceDictionary : (hasReads() ? readsHeader.getSequenceDictionary() : null);
+    }
+
+    /**
+     * @return sequence dictionary for the reference, or null if there is no reference available
+     */
+    public SAMSequenceDictionary getReferenceSequenceDictionary() {
+        return referenceDictionary;
+    }
+
+    /**
+     * @return header for the reads, or null if there are no reads available
+     */
+    public SAMFileHeader getHeaderForReads() {
```

Would it be better to raise an exception if there are no reads available?